lms/static/scripts/frontend_apps/components/test/AppRoot-test.js

```diff
@@ -50,14 +76,38 @@ describe('AppRoot', () => {
   ].forEach(({ config, appComponent }) => {
     it('launches correct app for "mode" config', () => {
       navigateTo(`/app/${config.mode}`);
-      const services = new Map();
-      const wrapper = mount(
-        <AppRoot initialConfig={config} services={services} />
-      );
+      const wrapper = renderAppRoot({ config, services: new Map() });
       assert.isTrue(wrapper.exists(appComponent));
     });
   });
 
+  it('loads config for file picker after route change', async () => {
+    // Mock FilePickerApp to observe config it receives.
+    let actualConfig;
+    function DummyFilePickerApp() {
+      actualConfig = useConfig();
+    }
+    DummyFilePickerApp.displayName = 'FilePickerApp';
+    $imports.$mock({
+      './FilePickerApp': DummyFilePickerApp,
+    });
+
+    // Render app initially in assignment view route.
+    const config = { mode: 'basic-lti-launch' };
+    navigateTo('/app/basic-lti-launch');
+    const wrapper = renderAppRoot({ config, services: new Map() });
+    assert.isTrue(wrapper.exists('BasicLTILaunchApp'));
+
+    // Navigate to assignment edit route.
+    const updatedConfig = { ...config, filePicker: {} };
+    fakeLoadFilePickerConfig.resolves(updatedConfig);
+    navigateTo('/app/content-item-selection');
+    await waitForElement(wrapper, 'FilePickerApp');
```

Comment:
Presumably this is the faked `DummyFilePickerApp`? Just checking my own understanding here.

Reply:
Yes, exactly.